File: baekjoon/제로.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
	int k;
	cin >> k;
	int a;
	int sum = 0;
	stack<int> st;
	for (int i = 0; i < k; ++i) {
		cin >> a;
		if (a == 0)
			st.pop();
		else
			st.push(a);
	}
	while (!st.empty()) {
		sum += st.top();
		st.pop();
	}
	cout << sum;

}

File: baekjoon/제로.java
import java.util.Scanner;
import java.util.Stack;

public class Main{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < n; i++) {
            int input = sc.nextInt();
            if (input == 0) {
                stack.pop();
            }else{
                stack.add(input);
            }
        }
        int sum=0;
        for (var value : stack) {
            sum+=value;
        }
        System.out.println(sum);
    }
}
